bluesky/callbacks/core.py

```diff
@@ -351,3 +352,46 @@ def start(self, doc):
     def _print(self, out_str):
         self._rows.append(out_str)
         self._out(out_str)
+
+
+class Retrieve(CallbackBase):
+    def __init__(self, root_map=None, handler_reg=None):
```

Comment:
Why is the handler registry optional?

Reply:
Again copied out of the databroker code. I can make it required though.

Reply:
Fair. But yes let's make it required.